src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -24,13 +24,17 @@
 class EmailValidator extends ConstraintValidator
 {
     /**
-     * @var bool
+     * @var string
      */
-    private $isStrict;
+    private $defaultMode;
 
-    public function __construct($strict = false)
+    public function __construct($strict = false, $defaultMode = 'html5')
     {
-        $this->isStrict = $strict;
+        $this->defaultMode = $defaultMode;
+
+        if (true === $strict) {
+            $this->defaultMode = Email::VALIDATION_MODE_STRICT;
```

Comment:
Not terribly happy with this. Suggestions?

Reply:
something like that yes (sorry for the CS)
```
    public function __construct($defaultMode = Email::VALIDATION_MODE_STRICT)
    {
switch ($defaultMode) {
case VALIDATION_MODE_HTML5:
case VALIDATION_MODE_STRICT:
case VALIDATION_MODE_LOOSE: break;
default: $defaultMode = $defaultMode ? VALIDATION_MODE_STRICT : VALIDATION_MODE_LOOSE;
}
```